opentracing/src/main/java/io/opentracing/noop/NoopSpan.java

```diff
@@ -27,22 +29,22 @@ public void finish() {
 
     @Override
     public Span setTag(String key, String value) {
-        return defaultSpan;
+        return instance;
     }
 
     @Override
     public Span setTag(String key, boolean value) {
-        return defaultSpan;
+        return instance;
     }
 
     @Override
     public Span setTag(String key, Number value) {
-        return defaultSpan;
+        return instance;
     }
 
     @Override
     public Span setBaggageItem(String key, String value) {
-        return defaultSpan;
+        return instance;
     }
 
     @Override
```

Comment:
FYI:  Java's convention is never to return (or to accept as an argument) null unless documented by apidoc or a @Nullable annotation. 

Nothing needs to be done here though, since this method's apidoc states this in the super class and the apidoc isn't overridden